docs/struct.md

```diff
@@ -2,9 +2,169 @@
+
+This is consistent with how C, C++, and many other languages store structs in memory.
+How's our struct-in-a-DataFrame different?
+
+Polars follows the Arrow protocol for structs, which means each field of the struct is stored in a Series, backed by chunks. Each chunk is contiguous in memory.
+That makes it so, for instance, assuming a single chunk, the address after 64 bits from the start of the first x field holds the x value for the next struct instance, not the y field of the same instance.
+It's also important to mention the chunks composing the Series are stored on the heap. In a scenario in which we have a single chunk for each field, this is how things would look like:
+
+![Diagram showing how Series backing struct fields are stored in memory](assets/structchunked_fields_memory_layout.png){ style="display: block; margin: 0 auto;" }
+
+Since we never modified the DataFrame after creating it, it is the case of our initial example, no more chunks were allocated.
+
+---
+
+Now that we have a better idea of how things work under the hood, let's jump to a practical plugin that takes a struct, and just prints the Series corresponding to each field - it'll return the same struct passed as input, with no alteration.
+
+First things first - this time we're gonna see somthing new. Polars does not allow us to write:
+
+```rust
+#[polars_expr(output_type=Struct)]
+fn print_fields(inputs: &[Series]) -> PolarsResult<Series> { ... }
+```
+
+The way we inform a struct Series is being returned is a bit cumbersome - we do so by defining a separate function:
+
+```rust
+fn struct_point_2d_output(input_fields: &[Field]) -> PolarsResult<Field> {
+    let field = &input_fields[0];
+    match field.dtype() {
+        DataType::Struct(fields) => {
+            Ok(Field::new("struct_point_2d".into(), DataType::Struct(fields.clone())))
+        }
+        // Assuming a controlled environment in which we know the input
+        _ => unreachable!(),
+    }
+}
+```
+
+Then using that function in our `polars_expr`, with a different "kwarg":
+
+```rust
+#[polars_expr(output_type_func=struct_point_2d_output)]
+fn print_series(inputs: &[Series]) -> PolarsResult<Series> {
+
+    let struct_ = inputs[0].struct_()?;
+    let fields = struct_.fields_as_series();
+
+    if fields.is_empty() {
+        return Ok(inputs[0].clone());
+    }
+
+    let fields = fields
+        .iter()
+        .map(|s| {
+            let s = s.clone();
+            println!("{:?}", s);
+            s
+        })
+        .collect::<Vec<_>>();
+
+    StructChunked::from_series(struct_.name().clone(), &fields)
+        .map(|ca| ca.into_series())
+}
+```
+
+This is a very basic, "do-nothing" example. For this reason, we're not gonna spend too much time here.
+Still, you're encouraged to register the plugin and try it for yourself.
+Now, let's look at something more interesting.
```

Comment:
I could expand this to show what the plugin prints, but that would make the chapter longer.
Let me know what's the preferred solution here.

Reply:
update: Mentioned it should be done as an exercise